pkg/apis/numaflow/v1alpha1/source.go

```diff
@@ -29,10 +29,37 @@ type Source struct {
 	HTTP *HTTPSource `json:"http,omitempty" protobuf:"bytes,3,opt,name=http"`
 	// +optional
 	Nats *NatsSource `json:"nats,omitempty" protobuf:"bytes,4,opt,name=nats"`
+	// +optional
+	UdTransformer *UDTransformer `json:"udtransformer,omitempty" protobuf:"bytes,5,opt,name=udtransformer"`
```

Comment:
Can we just use `transformer` in `json` and `protobuf`?

Reply:
Currently I am using `udtransformer` everywhere as I would like to keep the wording consistent across the code base.
I am ok with either `transformer` or `udtransfomer`.

`udtransformer` emphasize the fact that it's user defined while `transformer` distinguish itself from udf, udsink and future udsource.

I am also ok with your propose where we use `udtransformer` across the code base while we expose to user the `transformer` spec keyword.

@vigith @whynowy  Let me know which approach you guys prefer. Thanks!

Reply:
I am okay with `udtransformer`, makes more sense w.r.t maintainability.

Reply:
@vigith I had a offline discussion with Derek. He raised a point that `udtransformer` is not a valid word, which might confuse end users. Mind we use `transformer` in spec and `udtransformer` in code base?

Reply:
yes, please go ahead with `transformer` 😄